src/lib/input/input.ts

```diff
@@ -162,7 +168,8 @@ export class MatInput implements MatFormFieldControl<any>, OnChanges, OnDestroy,
               @Optional() protected _parentForm: NgForm,
               @Optional() protected _parentFormGroup: FormGroupDirective,
               private _defaultErrorStateMatcher: ErrorStateMatcher,
-              @Optional() @Self() @Inject(MAT_INPUT_VALUE_ACCESSOR) inputValueAccessor: any) {
+              @Optional() @Self() @Inject(MAT_INPUT_VALUE_ACCESSOR) inputValueAccessor: any,
+              @Optional() @Inject(PLATFORM_ID) platformId?: Object) {
```

Comment:
This class already injects `Platform` from cdk/platform, you could just use `!this._platform.isBrowser`

Reply:
`Platform` just checks if the `document` exists and has type `object`. This probably won't work now that universal gives us a domino document right?

We probably need to update the `Platform.isBrowser` logic, but I didn't want to do it in this PR.

Reply:
I'd be hesitant to change it now because it could be a breaking change

Reply:
That's why I left it out of this PR, but I do think we have to change it in the future. We need to audit the places where we're using this and check which ones are using it as a way to check if the `document` is defined and which ones really care that we're on the server. The latter checks are now incorrect due to Angular's changes. (The one I added in this PR for example, would be broken if I depended on `Platform.isBrowser`)